spec.html

```diff
@@ -34920,26 +34923,25 @@ <h1>Notation</h1>
         <h1>Runtime Semantics: CompilePattern</h1>
         <dl class="header">
           <dt>description</dt>
-          <dd>It returns an Abstract Closure that takes a String and a non-negative integer and returns a MatchResult.</dd>
+          <dd>It returns an Abstract Closure that takes a List of characters and a non-negative integer and returns a MatchResult.</dd>
         </dl>
         <emu-grammar>Pattern :: Disjunction</emu-grammar>
         <emu-alg>
           1. Let _m_ be CompileSubpattern of |Disjunction| with argument ~forward~.
-          1. Return a new Abstract Closure with parameters (_str_, _index_) that captures _m_ and performs the following steps when called:
-            1. Assert: Type(_str_) is String.
-            1. Assert: _index_ is a non-negative integer which is &le; the length of _str_.
-            1. If _Unicode_ is *true*, let _Input_ be ! StringToCodePoints(_str_). Otherwise, let _Input_ be a List whose elements are the code units that are the elements of _str_. _Input_ will be used throughout the algorithms in <emu-xref href="#sec-pattern-semantics"></emu-xref>. Each element of _Input_ is considered to be a character.
+          1. Return a new Abstract Closure with parameters (_input_, _index_) that captures _m_ and performs the following steps when called:
+            1. Assert: Type(_input_) is a List of characters.
+            1. Assert: _index_ is a non-negative integer which is &le; the length of _input_.
```

Comment:
```suggestion
            1. Assert: _index_ is a non-negative integer which is &le; the number of characters in _input_.
```
Generally, when an algorithm refers to `the length of _X_`, `_X_` is a String. For Lists, we usually use `the number of elements in _X_`, but in this case it can be more specific.